centaur/src/main/scala/centaur/test/Test.scala

```diff
@@ -780,13 +780,39 @@ object Operations extends StrictLogging {
       }
 
       def checkArchived(): IO[Boolean] = for {
-        archiveStatus <- CentaurCromwellClient.archiveStatus(workflowId)
+        archiveStatus <- CentaurCromwellClient.archiveStatus(submittedWorkflow.id)
         isArchived <- validateMetadataArchiveStatus(archiveStatus)
       } yield isArchived
 
+      def validateMetadataSourceArchived(): IO[Boolean]  = for {
+        metadataSource <- CentaurCromwellClient.metadataWithId(submittedWorkflow.id).map(_.asFlat.stringifyValues.get("metadataSource"))
+        isMetadataSourceArchived <- {
+          if (metadataSource == Option(JsString("Archived"))) {
+            IO.pure(true)
+          } else if (metadataSource == Option(JsString("Unarchived"))) {
+            IO.pure(false)
+          } else {
+            throw CentaurTestException(
```

Comment:
This should be wrapped with `IO.raiseError` (it's currently just throwing the exception and not returning an `IO` instance).